Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3541,9 +3541,15 @@ - (void)updateGroupModelTo:(TSGroupModel *)newGroupModel successCompletion:(void
 
         groupThread.groupModel = newGroupModel;
         [groupThread saveWithTransaction:transaction];
-        message = [[TSOutgoingMessage alloc] initWithTimestamp:[NSDate ows_millisecondTimeStamp]
-                                                      inThread:groupThread
-                                              groupMetaMessage:TSGroupMessageUpdate];
+        message = [[TSOutgoingMessage alloc] initOutgoingMessageWithTimestamp:[NSDate ows_millisecondTimeStamp]
```

Comment:
While working on this, I decided to address a long-standing issue.  The interactions have a deep inheritance hierarchy.  There were problems:

* It was easy to use the wrong initializer by accident, especially when using auto-complete.
* It was hard to tell at a glance if the wrong initializer had been used.
* We were using designated initializers in an inconsistent way.

I've tried to remedy this by ensuring that most interaction classes have "1 clearly correct and designated" initializer.  Each subclass marks its superclass' initializer as "unavailable".  The correct initializer has a visual distinct name based on the class name.

The biggest downside to this approach is that creating interactions is more verbose as you have to specify every relevant parameter.  Open to feedback!

Reply:
Yup, we should probably have "simple" and "omnibus" flavors of these initializers.   But a factory might help too.